multiplexer/main.go

```diff
@@ -25,21 +49,56 @@ func main() {
 			return
 		}
 
-		for _, payLoad := range urls.List {
-			resp, err := client.Get(payLoad)
-			if err != nil {
-				w.Write([]byte(err.Error()))
-			}
-			defer resp.Body.Close()
+		if len(urls.List) > 20 {
+			w.Write([]byte("amount of url's exceeded limit"))
+			return
+		}
 
-			b, err := ioutil.ReadAll(resp.Body)
-			if err != nil {
-				w.Write([]byte(err.Error()))
-			}
+		for _, url := range urls.List {
+			wg.Add(1)
+			go func(u string) {
+				mu.RLock()
+				content, err := getContent(&client, u)
+				mu.RUnlock()
+				
+				if err != nil {
+					w.Write([]byte(err.Error()))
+				}
 
-			w.Write(b)
+				mu.Lock()
+				w.Write(content)
+				mu.Unlock()
+
+				wg.Done()
+			}(url)
 		}
+		wg.Wait()
 	})
 
-	log.Println(http.ListenAndServe(":8080", nil))
+	srv := &http.Server{
+		Addr:    ":8080",
+		Handler: nil,
+	}
+
+	go func() {
+		if err := srv.ListenAndServe(); err != nil && err != http.ErrServerClosed {
```

Comment:
Think i should remove this because `srv.ListenAndServe()` 

> shuts down the server without interrupting any active connections. 

And in **Limitations #5**, i was told to 

>  stop all work & requests that was started in such client request

Whatcha think?